elisp/defs.bzl

```diff
@@ -408,6 +427,34 @@ To add a load path entry for the current package, specify `.` here.""",
             doc = "List of `elisp_library` dependencies.",
             providers = [EmacsLispInfo],
         ),
+        "enable_package": attr.bool(
+            doc = """Enable generation of package.el package for this library.
+This value is forced to False if testonly is True.""",
+            default = True,
+        ),
+        "emacs_package_name": attr.string(
+            doc = """The name used for the package.el package.
+This attribute is ignored if enable_package is False.
+Otherwise, srcs should contain a package description file `<name>-pkg.el`.
+If there is no such package description file, then srcs must contain a file
+`<name>.el` containing the appropriate package headers.
+
+If there is only one file in srcs, then the default value is the file basename
+with the .el suffix removed.  Otherwise, the default is the target label name,
+with underscores replaced with dashes.""",
+        ),
+        "_gen_pkg_el": attr.label(
+            default = "//elisp:gen-pkg-el.el",
```

Comment:
Instead of loading the source files here, could you use `elisp_binary` rules or (if that doesn't work due to dependency cycles) the `bootstrap` rule and a compile file, like `compile.elc`? Loading compiled files should be a bit more robust and faster.

Reply:
Done